pkg/apis/storage/types.go

```diff
@@ -337,6 +337,35 @@ type CSIDriverSpec struct {
 	//
 	// +optional
 	StorageCapacity *bool
+
+	// ServiceAccountTokens indicates the csi driver needs pods' service account
+	// tokens it is mounting volume for to do necessary authentication. Kubelet
+	// will pass the tokens in VolumeContext in the CSI NodePublishVolume calls.
+	// The CSI driver should parse and validate the following VolumeConext:
+	// "csi.storage.k8s.io/serviceAccount.tokens": {
+	//   'audience-0': {
+	//     'token': token-0,
+	//     'expiry': expiry-0,
+	//   },
+	//   // If audience is not specified, the output key will be ''
+	//   '': {
+	//     'token': token,
+	//     'expiry': expiry,
+	//   },
+	//   ...
+	// }
+	//
+	// Note: Audience in each ServiceAccountToken should be different and at
+	// most one token can ignore Audience.
+	//
+	// +optional
+	// +listType=set
+	ServiceAccountTokens []ServiceAccountToken
+
+	// RequiresRemount indicates the csi driver wants `NodePublishVolume` being
+	// periodically called to reflect any possible change in the mounted volume.
+	// +optional
+	RequiresRemount *bool
```

Comment:
To align with CSI terminology, I would call this something like `RequiresRepublish`. I know the code name is different but I don't think we need to expose internal code function names in the API.

Reply:
I think we also need to make clear here that drivers would need to repopulate the content, but not actually do a remount. Because CRI has already bind mounted the mount point, and a new mount point cannot be updated without a container restart.

And in that case, do we also need to pass in an attribute to the driver to tell them that this is a republish vs the 1st time mounting? cc @jsafrane for thoughts.

Reply:
added a note.